pkg/vault/test/authorizer/TimelockAuthorizerPermissions.test.ts

```diff
@@ -0,0 +1,802 @@
+
+        const receipt = await authorizer.cancel(id, { from: getSender() });
+        expectEvent.inReceipt(await receipt.wait(), 'ExecutionCancelled', { scheduledExecutionId: id });
+      });
+
+      it('can be executed after the expected delay', async () => {
+        const id = await authorizer.scheduleRevokePermission(ACTION_1, user, WHERE_1, [], { from: getSender() });
+
+        await advanceTime(delay);
+        const receipt = await authorizer.execute(id);
+        expectEvent.inReceipt(await receipt.wait(), 'ExecutionExecuted', { scheduledExecutionId: id });
+      });
+
+      it('revokes the permission when executed', async () => {
+        // grant the premission first
+        await authorizer.grantPermission(ACTION_1, user, WHERE_1, { from: root });
+
+        expect(await authorizer.hasPermission(ACTION_1, user, WHERE_1)).to.be.equal(true);
+
+        const id = await authorizer.scheduleRevokePermission(ACTION_1, user, WHERE_1, [], { from: getSender() });
+
+        await advanceTime(delay);
+        await authorizer.execute(id);
+
+        expect(await authorizer.hasPermission(ACTION_1, user, WHERE_1)).to.be.equal(false);
+      });
+
+      it('does not revoke any other permissions when executed', async () => {
+        const id = await authorizer.scheduleRevokePermission(ACTION_1, user, WHERE_1, [], { from: getSender() });
+
+        await advanceTime(delay);
+        await authorizer.execute(id);
+
+        expect(await authorizer.hasPermission(ACTION_3, user, WHERE_1)).to.be.equal(false);
+        expect(await authorizer.hasPermission(ACTION_2, user, WHERE_2)).to.be.equal(false);
+      });
+
+      it('emits an event', async () => {
+        const receipt = await authorizer.instance
+          .connect(getSender())
+          .scheduleRevokePermission(ACTION_1, user.address, WHERE_1, []);
+
+        expectEvent.inReceipt(await receipt.wait(), 'RevokePermissionScheduled', {
+          actionId: ACTION_1,
+          account: user.address,
+          where: WHERE_1,
+        });
+      });
+    }
+
+    context('when the sender is root', () => {
+      itScheduleRevokePermissionCorrectly(() => root);
+    });
+
+    context('when the sender is revoker', () => {
+      sharedBeforeEach('makes a revoker', async () => {
+        await authorizer.addRevoker(revoker, EVERYWHERE, { from: root });
+      });
+      itScheduleRevokePermissionCorrectly(() => revoker);
+    });
+  });
+
+  describe('renouncePermission', () => {
+    context('when the sender does not have the permission', () => {
+      context('when renouncing the permission for a specific contract', () => {
+        it('ignores the request and still cannot perform the requested action everywhere', async () => {
+          await expect(authorizer.renouncePermission(ACTION_1, WHERE_1, { from: user })).not.to.be.reverted;
+
+          expect(await authorizer.canPerform(ACTION_1, user, EVERYWHERE)).to.be.false;
+        });
```

Comment:
I'd also consider reverting here

Reply:
I agree but let's make it a separate PR. This one is big enough.

Reply:
Created https://github.com/balancer/balancer-v2-monorepo/issues/2379